src/Uno.UWP/Storage/Pickers/FileOpenPicker.Android.cs

```diff
@@ -79,8 +96,20 @@ private async Task<FilePickerSelectedFilesArray> PickFilesAsync(bool multiple, C
 						continue;
 					}
 					var file = StorageFile.GetFromSafUri(item.Uri);
-					files.Add(file);
-				}
+					files.Add(file);
+
+					// for PickMultipleFilesAsync(), we preserve (existing) path of last selected file
+					if (!string.IsNullOrEmpty(file.Path))
+					{
+						ApplicationData.Current.LocalSettings.Values[settingName] = file.Path;
+						wasPath = true;
+					}
+				}
+
+				if(!wasPath)
+                {	// if we have no path in any of files, remove setting - next call to Picker will not have InitialDir
+					ApplicationData.Current.LocalSettings.Values.Remove(settingName);
+				}

```

Comment:
Can you adjust the indentation?

Reply:
On my side it looks good. But I copied leading white spaces from line 112 to line 110, maybe it helps.